### PR TITLE
Add custom apps translation scripts and image path for consistency

### LINKED ISSUE #40897: Custom apps translation and image asset handling is not consistent with core apps
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
- Custom app translation scripts are not delivered at all
  - For files which is a core app, I see a `GET apps/files/l10n/fr.js`
  - But for a custom app, this is not there
- I can replace core app translations via theme but not for custom apps
- I can replace core app images via theme but not for custom apps

**Describe the solution you'd like**
- Consistent behaviour for both core and custom apps

**Describe alternatives you've considered**
- N/A

**Additional context**
- N/A

### PR DESCRIPTION
**Pull request by akhil1508**
* Resolves: https://github.com/nextcloud/server/issues/40897

## Summary
- Add checks for custom app translations and images

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/Template/JSResourceLocator.php [hunk 1]
```diff
@@ -62,6 +62,8 @@ public function doFind($script) {
 			$found += $this->appendScriptIfExist($this->serverroot, $theme_dir.$script);
 			$found += $this->appendScriptIfExist($this->serverroot, 'apps/'.$script);
 			$found += $this->appendScriptIfExist($this->serverroot, $theme_dir.'apps/'.$script);
+			$found += $this->appendScriptIfExist($this->serverroot, 'custom_apps/'.$script);
```
Comment (kesselb) [inline lib/private/Template/JSResourceLocator.php:65]: It's possible to pick a different name than custom_apps and therefore this approach is wrong.

**Review by kesselb — CHANGES_REQUESTED** [review]
I'm unsure about this change in general.
I have plenty of apps in apps-extra and see the requests for the translation files.

**Comment by akhil1508** [comment]
> I'm unsure about this change in general.
> I have plenty of apps in apps-extra and see the requests for the translation files.

For me, `occ app:install calendar`  installed it in `custom_apps` folder. I don't see the requests for `custom_apps/calendar/l10n/$lang.js` for any language. Nor if I modify via theme..

File: lib/private/Template/JSResourceLocator.php (hunk 1 above)
Comment (akhil1508) [reply lib/private/Template/JSResourceLocator.php]: Yes, my bad. Different approach now :)

File: lib/private/Template/JSResourceLocator.php [hunk 2]
```diff
@@ -60,8 +60,13 @@ public function doFind($script) {
 			$found += $this->appendScriptIfExist($this->serverroot, $theme_dir.'core/'.$script);
 			$found += $this->appendScriptIfExist($this->serverroot, $script);
 			$found += $this->appendScriptIfExist($this->serverroot, $theme_dir.$script);
-			$found += $this->appendScriptIfExist($this->serverroot, 'apps/'.$script);
-			$found += $this->appendScriptIfExist($this->serverroot, $theme_dir.'apps/'.$script);
+
+			foreach (\OC::$APPSROOTS AS $appRoot) {
```
Comment (kesselb) [inline lib/private/Template/JSResourceLocator.php:64]:
```suggestion
			foreach (\OC::$APPSROOTS as $appRoot) {
```

**Comment by kesselb** [comment]
@juliushaertl @skjnldsv I'm totally unsure if the patch makes sense, we need it, etc .. :see_no_evil: 

However, I dislike the ability to modify files from apps via a theme.

**Comment by juliushaertl** [comment]
I'm all in for dropping the theme generally once we have proper ways to achieve its mainly used functionality  otherwise https://github.com/nextcloud/server/issues/27433

I think for now the change makes sense, to ensure that the legacy theme mechanism works as expected with custom app paths.

**Review by juliusknorr — APPROVED** [review]
Would be good to squash the commits into one, otherwise this seems good.

**Comment by akhil1508** [comment]
> Would be good to squash the commits into one

@juliushaertl Squashed

**Review by susnux — APPROVED** [review]

**Comment by juliushaertl** [comment]
Failures seem unrelated.

**Pull request merged**